CONTRIBUTING.md

```diff
@@ -170,6 +170,27 @@ for coding advice). The code must adhere to the following robustness principles
 are important for software that runs autonomously and continuously without direct
 interaction with a human (such as this Collector).
 
+### Naming convention
+
+To keep naming patterns consistent across the project, naming patterns are enforced to make intent clear by:
+
+- Methods that return a variable that uses the zero value or values provided via the method MUST have the prefix `New`. For example:
+  - `func NewKinesisExporter(kpl aws.KinesisProducerLibrary)` allocates a variable that uses
```

Comment:
Can you use [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt) for all entries here?

- `func NewKinesisExporter(kpl aws.KinesisProducerLibrary)` shall allocate...`
- `func NewKeyValueBuilder()` may allocate internal varialbes to a safe zero value (is this really a "may" according to that RFC?)